### PR TITLE
Fix reading files with SD1700 in company descriptive date field

### PR DESCRIPTION
**Pull request by henriquegasques**
### Problem

Reading a file with with 'SDxxxx' in the company descriptive date field of the batch header was trying to parse the string to a Date, causing it to be `nil`

### Solution

Return the actual value if it doesn't parse to a Date, so it can be formatted when building the batch header.

### CODE REVIEW

**Comment by henriquegasques** [comment]
Hi @jm81, if you can, please take a look at this proposed fix and let me know what you think.

I know you are not actively maintaining this project, but it would be nice to have this merged if possible. Also, what would be the steps to become a maintainer? I'm currently using this gem at work, and a few colleagues and I would like to help maintain it.

**Comment by adamdecaf** [comment]
`SDHHMM` could be parsed as `HH` (hour) and `MM` (minute) to represent same-day ach.

**Comment by henriquegasques** [comment]
@adamdecaf That makes complete sense, specially to keep backwards compatibility since `DateTime` is also a `Date`. Done here 1a5b7fb

File: lib/ach/ach_file.rb [hunk 1]
```diff
@@ -160,6 +160,15 @@ def parse data
     def parse_descriptive_date(date_string)
       return if date_string.strip.empty?
 
+      date_time =
+        case date_string
+        when 'SD1300' then Date.today + 13.0/24
+        when 'SD1700' then Date.today + 17.0/24
+        when 'SD1800' then Date.today + 18.0/24
+        end
```
Comment (adamdecaf) [inline lib/ach/ach_file.rb:168]: I'm rusty on ruby, but SDHHMM is hours and minutes. There could be any value from SD0001 to SD2359 in practice.

Comment (henriquegasques) [reply lib/ach/ach_file.rb]: Refactored using a regex (hopefully I don't have two problems now)

**Comment by jm81** [comment]
@henriquegasques The change makes sense to me. Since it looks like you might still be working on it, I haven't merged yet. I am inviting you as a collaborator so you can merge whenever if it's ready and you don't want to wait on me. I'd be happy to add @adamdecaf or anyone else as a collaborator too.

As far as taking over or helping out with maintaining, that would be awesome. I can give collaborator access as needed or if you'd like to put the official repo on another account, I can just update the README to point there. And then I could add you as an owner of the gem on rubygems.org. If you have an account there and want me to so, just let me know your email or handle for rubygems. But you can also just let me know when it's ready for a new gem version to be pushed and I can do so though it might take me a couple days.

**Review by adamdecaf — APPROVED** [review]
This handles SD values a lot better now.

**Comment by adamdecaf** [comment]
I could be a collaborator here, but I've already [got an ACH project to maintain](https://github.com/moov-io/ach). I just watch this repository to see what other projects are up to and help out where I can.

**Comment by jm81** [comment]
I've pushed version 0.6.1 to rubygems.org

**Pull request merged**